include/dirac_quda.h

```diff
@@ -2238,6 +2250,48 @@ namespace quda {
     }
   };
 
+  /**
+     Gloms onto a DiracMatrix and provides an operator() for its G5M method
+  */
+  class DiracG5M : public DiracMatrix
+  {
+
+  public:
+    DiracG5M(const Dirac &d) : DiracMatrix(d) { }
+    DiracG5M(const Dirac *d) : DiracMatrix(d) { }
+
+    void operator()(ColorSpinorField &out, const ColorSpinorField &in) const
+    {
+      dirac->M(out, in);
+      if (shift != 0.0) blas::axpy(shift, const_cast<ColorSpinorField &>(in), out);
+      gamma5(out, out);
+    }
+
+    void operator()(ColorSpinorField &out, const ColorSpinorField &in, ColorSpinorField &tmp) const
+    {
+      dirac->tmp1 = &tmp;
```

Comment:
This needs the `reset` logic present, for ex, in `DiracMdag` (which I added recently). This prevents an existing `dirac->tmp1`, etc, from being overriden if it's already set, which happens within multigrid. (I know this PR doesn't support gamma5 w/coarse ops yet, so this is more being proactive.)

Reply:
https://github.com/lattice/quda/commit/78d633b6807d1ae4f9d33583ae7c30b145e3fe65